dagshub/data_engine/model/query.py

```diff
@@ -183,31 +191,47 @@ def _serialize_node(node: Node, tree: Tree) -> Dict:
             value = node.data["value"]
             as_of = node.data.get("as_of")
 
-            value_type = metadataTypeLookup[type(value)].value
-            if type(value) is bytes:
-                # TODO: this will need to probably be changed when we allow actual binary field comparisons
-                value = value.decode("utf-8")
+            if query_op in [FieldFilterOperand.YEAR,
+                            FieldFilterOperand.MONTH,
+                            FieldFilterOperand.DAY,
+                            FieldFilterOperand.TIMEOFDAY]:
+                res = {
+                    "filter": {
+                        "key": key,
+                        "value": value if query_op is FieldFilterOperand.TIMEOFDAY else 0,
+                        "valueRange": 0 if query_op is FieldFilterOperand.TIMEOFDAY else value,
```

Comment:
Is there a point to doing the valueRange thing if it can be instead represented as multiple ORs?
I don't know how the frontend would go about handling all of this though, but adding a new field in the filter just for this seems strange to me

Reply:
we could, but since in the UI there is an option to mark multiple days, its more convenient to have a single filter built at once